File: c/4.c
#include <stdio.h>

int palin(int n)
{
	return n / 100000 == n % 10 &&
		n % 100000 / 10000 == n % 100 / 10 &&
		n % 10000 / 1000 == n % 1000 / 100;
}

int main(void)
{
	int m, n, prod, max = 0;

	for (m = 900; m < 1000; ++m)
		for (n = 900; n < 1000; prod = m * ++n)
			if (prod > max && palin(prod))
				max = prod;

	printf("%d\n", max);
	return 0;
}

File: js/4.js
palin = function(n) {
    s = n.toString().split('');
    return s.join('') == s.reverse().join('');
}

max = 0;

for (a = 900; a < 1000; ++a) {
    for (b = 900; b < 1000; ++b) {
        prod = a * b;
        if (prod > max && palin(prod))
            max = prod;
    }
}

console.log(max);
